.github/workflows/core.yml

```diff
@@ -103,9 +103,12 @@ jobs:
         with:
           variant: ${{ matrix.install-variant }}
           python-version: ${{ matrix.python-version }}
+      - name: checkout current repo
+        uses: actions/checkout@v4
+      - name: debug run install current repo
+        run : pip install .
       #cypress starts here
       - name: Check out files needed to run cypress tests
-        if: matrix.install-variant != 'dev'
```

Comment:
After running some tests on my fork, I believe this is the cause of the `ModuleNotFoundError` with the `dev` install variant. The error was introduced accidentally with the removal of the conditional `if: matrix.install-variant != 'dev`, which caused `actions/checkout` to do a partial clone (which only fetches the files listed there, as opposed to the whole repo) which overwrites the Python package that had been installed in editable mode by `.github/actions/install`.

These should restore the correct behavior:

1) Add back the line `if: matrix.install-variant != 'dev'`
2) Remove these 2 extra steps (lines 106-109): 
![image](https://github.com/IDAES/idaes-ui/assets/48035537/3dd44c8f-c6b4-43f2-9e87-f128d96d31c5)